target/linux/qualcommax/ipq807x/base-files/etc/init.d/smp_affinity

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh /etc/rc.common
+
+START=99
+
+enable_affinity_ipq807x(){
+	set_affinity() {
+		irq=$(awk "/$1/{ print substr(\$1, 1, length(\$1)-1); exit }" /proc/interrupts)
+		[ -n "$irq" ] && echo $2 > /proc/irq/$irq/smp_affinity
+	}
+
+	# assign 4 rx interrupts to each core
+	set_affinity 'reo2host-destination-ring1' 1
+	set_affinity 'reo2host-destination-ring2' 2
+	set_affinity 'reo2host-destination-ring3' 4
+	set_affinity 'reo2host-destination-ring4' 8
+
+	# assign 3 tcl completions to last 3 CPUs
+	set_affinity 'wbm2host-tx-completions-ring1' 2
+	set_affinity 'wbm2host-tx-completions-ring2' 4
+	set_affinity 'wbm2host-tx-completions-ring3' 8
+
+	# assign 3 ppdu mac interrupts to last 3 cores
+	set_affinity 'ppdu-end-interrupts-mac1' 2
+	set_affinity 'ppdu-end-interrupts-mac2' 4
+	set_affinity 'ppdu-end-interrupts-mac3' 8
+
+	# assign lan/wan to core 4
+	set_affinity 'edma_txcmpl' 8
+	set_affinity 'edma_rxfill' 8
+	set_affinity 'edma_rxdesc' 8
+	set_affinity 'edma_misc' 8
```

Comment:
Hi! As curiosity, why are all this four IRQs assigned to the same core? It enhances performance being all of them in the same core? Until now I used each of them for different cores.

Thanks for your work with IPQ807!!

Reply:
No idea really, this has been floating around the forums for a long time and I was asked multiple time if this can be merged so everybody benefits from it by default.

Other than using it on my AP I haven't really investigated the decisions in it

Reply:
I have seen on other SoC-s that IRQ-s for a certain MAC shouldn't be split on different cores as it can cause rather weird issues.

Reply:
Then perfect. They talk about a crash in the network. At least as first version this PR is ok then, seems safest.
Thanks for the explanation!

Reply:
I have been using this on my DL-WRX36 for a long time without  a problem so all in favour of this PR